### PR TITLE
doc: Add an example

### PR DESCRIPTION
**Pull request by larseggert**

### CODE REVIEW

**Comment by mxinden** [comment]
It is failing on the doc test introduced in this pull request. I will take a look why that is the case.

**Comment by mxinden** [comment]
@larseggert can you try the following:

``` diff
modified   .github/workflows/check.yml
@@ -88,7 +88,7 @@ jobs:
 
           for sanitizer in $SANITIZERS; do
             echo "Running tests with $sanitizer sanitizer..."
-            RUSTFLAGS="-Z sanitizer=$sanitizer" cargo +nightly test -Z build-std --target $TARGET
+            RUSTFLAGS="-Z sanitizer=$sanitizer" RUSTDOCFLAGS="-Z sanitizer=$sanitizer" cargo +nightly test -Z build-std --target $TARGET
           done
```

**Comment by mxinden** [comment]
```
test src/lib.rs - get_interface_mtu (line 32) ... ok
```

Seems to work.

File: .rustfmt.toml
Comment (mxinden) [inline .rustfmt.toml:1]: :pray:

File: src/lib.rs [hunk 1]
```diff
@@ -20,7 +20,24 @@ fn default_result<T>() -> Result<T, Error> {
     ))
 }
 
-/// Return the MTU of the interface that is used to reach the given remote socket address.
+/// Return the maximum transmission unit (MTU) of the local network interface towards the
+/// destination `SocketAddr` given in `remote`.
```
Comment (mxinden) [inline src/lib.rs:24]:
```suggestion
/// destination [`SocketAddr`] given in `remote`.
```

File: src/lib.rs [hunk 2]
```diff
@@ -20,7 +20,24 @@ fn default_result<T>() -> Result<T, Error> {
...
+///
+/// ```
+/// use std::net::ToSocketAddrs;
+///
+/// use mtu::get_interface_mtu;
+///
+/// let saddr = "localhost:443".to_socket_addrs().unwrap().next().unwrap();
+/// let mtu = get_interface_mtu(&saddr).unwrap();
+/// println!("MTU towards {:?} is {}", saddr, mtu);
+/// ```
```
Comment (mxinden) [inline src/lib.rs:40]:
```suggestion
/// ```
/// # use std::net::ToSocketAddrs;
/// # use mtu::get_interface_mtu;
/// let saddr = "127.0.0.1:443".parse().unwrap();
/// let mtu = get_interface_mtu(&saddr).unwrap();
/// println!("MTU towards {:?} is {}", saddr, mtu);
/// ```
```

Just some nits

**Review by mxinden — APPROVED** [review]

File: src/lib.rs [hunk 3]
```diff
@@ -20,7 +20,22 @@ fn default_result<T>() -> Result<T, Error> {
...
+///
+/// The returned MTU may exceed the maximum IP packet size of 65,535 bytes on some
+/// platforms for some remote destinations. (For example, loopback destinations on
+/// Windows.)
+///
+/// # Examples
+///
+/// ```
+/// # use std::net::ToSocketAddrs;
+/// # use mtu::get_interface_mtu;
```
Comment (larseggert) [inline src/lib.rs:34]: @mxinden should I just remove these lines?

Comment (larseggert) [reply src/lib.rs:34]: Well, at least the `ToSocketAddrs` one.

Comment (mxinden) [reply src/lib.rs:34]: Adding `# <some rust code>` simply hides `<some rust code>` when rendering the docs. They are still necessary though.

Comment (mxinden) [reply src/lib.rs:34]: https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#hiding-portions-of-the-example

**Comment by mxinden** [comment]
I don't think the latest commit will compile. See https://github.com/mozilla/mtu/pull/6#discussion_r1741020895 in case it is already hidden by GitHub UI.

**Pull request merged**